txpipe/extensions/twopoint_scia.py

```diff
@@ -0,0 +1,502 @@
+            # the other two are together below
+            if d.corr_type == XI:
+                xip = d.object.xip
+                xim = d.object.xim
+                xiperr = np.sqrt(d.object.varxip)
+                ximerr = np.sqrt(d.object.varxim)
+                n = len(xip)
+                # add all the data points to the sacc
+                for i in range(n):
+                    S.add_data_point(XIP, (tracer1,tracer2), xip[i],
+                        theta=theta[i], error=xiperr[i], npair=npair[i], weight= weight[i])
+                    S.add_data_point(XIM, (tracer1,tracer2), xim[i],
+                        theta=theta[i], error=ximerr[i], npair=npair[i], weight= weight[i])
+            else:
+                xi = d.object.xi
+                err = np.sqrt(d.object.varxi)
+                n = len(xi)
+                for i in range(n):
+                    S.add_data_point(d.corr_type, (tracer1,tracer2), xi[i],
+                        theta=theta[i], error=err[i], weight=weight[i])
+
+        # Add the covariance.  There are several different jackknife approaches
+        # available - see the treecorr docs
+        cov = treecorr.estimate_multi_cov(comb, self.config['var_method'])
+        S.add_covariance(cov)
+
+        # Our data points may currently be in any order depending on which processes
+        # ran which calculations.  Re-order them.
+        S.to_canonical_order()
+
+        # Finally, save the output to Sacc file
+        S.save_fits(self.get_output('twopoint_data_SCIA'), overwrite=True)
+
+        if self.config['do_shear_pos'] == True:
+            comb = []
+            for d in results:
+                tracer1 = f'source_{d.i}' 
+                tracer2 = f'source_{d.j}'    
+                
+                if d.corr_type == GAMMAT:
+                    theta = np.exp(d.object.meanlogr)
+                    npair = d.object.npairs
+                    weight = d.object.weight
+                    xi_x = d.object.xi_im
+                    covX = d.object.estimate_cov('shot')
+                    comb.append(covX)
+                    err = np.sqrt(np.diag(covX))
+                    n = len(xi_x)
+                    for i in range(n):
+                        S2.add_data_point(GAMMAX, (tracer1,tracer2), xi_x[i],
+                            theta=theta[i], error=err[i], weight=weight[i])
+                if d.corr_type == GAMMATS:
+                    theta = np.exp(d.object.meanlogr)
+                    npair = d.object.npairs
+                    weight = d.object.weight
+                    xi_x = d.object.xi_im
+                    covX = d.object.estimate_cov('shot')
+                    comb.append(covX)
+                    err = np.sqrt(np.diag(covX))
+                    n = len(xi_x)
+                    for i in range(n):
+                        S2.add_data_point(GAMMAXS, (tracer1,tracer2), xi_x[i],
+                            theta=theta[i], error=err[i], weight=weight[i])
```

Comment:
Should there be any difference for the two blocks of code of gammaT vs gammaTS? It is not obvious to me how the selection function goes into the second one here (is it necessary when dealing with the covariances?)

Reply:
This is purely writing the data into a sacc data file. Notice the GAMMAX and GAMMAXS in lines 485 and 497. They mean we save them as different names.